cartridge/rpc.lua

```diff
@@ -76,17 +78,40 @@ local function get_candidates(role_name, opts)
     if opts.healthy_only == nil then
         opts.healthy_only = true
     end
+    if opts.retry_timeout == nil then
+        opts.retry_timeout = 3
+    end
+    if opts.retry_delay == nil then
+        opts.retry_delay = 0.1
+    end
 
     checks('string', {
         leader_only = '?boolean',
-        healthy_only = '?boolean'
+        healthy_only = '?boolean',
+        retry_delay = '?number',
+        retry_timeout = '?number',
     })
 
     local topology_cfg = confapplier.get_readonly('topology')
     if topology_cfg == nil then
         return {}
     end
 
+    local end_time = fiber.time() + opts.retry_timeout
+    local patch_in_progress
+    repeat
+        patch_in_progress = (twophase_vars.locks or {})['clusterwide']
+        if not patch_in_progress then
+            break
+        else
+            fiber.sleep(opts.retry_delay)
+        end
+    until fiber.time() <= end_time
+
+    if patch_in_progress then
+        return nil, RemoteCallError:new('Twophase commit is in progress, try again later')
```

Comment:
All TDG tests fail after this patch. Please don't do that.

Reply:
I pushed my "optimistic" approach https://github.com/tarantool/cartridge/pull/1724

I have not tested it against TDG but seems it doesn't ban anything just add some timeout to wait when config commit will be finished.